rust/private/rustc.bzl

```diff
@@ -2272,34 +2427,35 @@ def _add_native_link_flags(args, dep_info, linkstamp_outs, ambiguous_libs, crate
 
     args.add_all(linkstamp_outs, before_each = "-C", format_each = "link-args=%s")
 
-    if crate_type in ["dylib", "cdylib"]:
-        # For shared libraries we want to link C++ runtime library dynamically
-        # (for example libstdc++.so or libc++.so).
-        args.add_all(
-            cc_toolchain.dynamic_runtime_lib(feature_configuration = feature_configuration),
-            map_each = _get_dirname,
-            format_each = "-Lnative=%s",
-        )
-        if include_link_flags:
+    if cc_toolchain:
+        if crate_type in ["dylib", "cdylib"]:
+            # For shared libraries we want to link C++ runtime library dynamically
+            # (for example libstdc++.so or libc++.so).
             args.add_all(
                 cc_toolchain.dynamic_runtime_lib(feature_configuration = feature_configuration),
-                map_each = get_lib_name,
-                format_each = "-ldylib=%s",
+                map_each = _get_dirname,
+                format_each = "-Lnative=%s",
             )
-    else:
-        # For all other crate types we want to link C++ runtime library statically
-        # (for example libstdc++.a or libc++.a).
-        args.add_all(
-            cc_toolchain.static_runtime_lib(feature_configuration = feature_configuration),
-            map_each = _get_dirname,
-            format_each = "-Lnative=%s",
-        )
-        if include_link_flags:
+            if include_link_flags:
+                args.add_all(
+                    cc_toolchain.dynamic_runtime_lib(feature_configuration = feature_configuration),
+                    map_each = get_lib_name,
+                    format_each = "-ldylib=%s",
+                )
+        else:
+            # For all other crate types we want to link C++ runtime library statically
+            # (for example libstdc++.a or libc++.a).
             args.add_all(
                 cc_toolchain.static_runtime_lib(feature_configuration = feature_configuration),
-                map_each = get_lib_name,
-                format_each = "-lstatic=%s",
+                map_each = _get_dirname,
```

Comment:
do we actually need to use these -L or can we just pass `static_runtime_lib` directly so it expands to the actual libs

Reply:
We could probably pass `static_runtime_lib` but my changes here are mostly cleanup so I wouldn't want to include that in this PR.